### DISCUSSION TITLE
Port for .NET nanoFramework

### DISCUSSION THREAD

**Discussion by josesimoes** [micjahn/ZXing.Net#397] [General]
Hey!

Was looking at this library and thinking that it would be nice to have a port for it to run on [.NET nanoFramework](https://github.com/nanoframework). Considering the resource constrain of the platforms we run in it would probably be better to offer a "reader only" version and another "generator only".

Would you be interested in "welcoming" this or we are better forking it and take it from there? 😉

**Comment by micjahn** [comment]
A while ago I made a version for the .Net Micro Framework.
https://github.com/micjahn/ZXing.Net-NETMF
Perhaps that a good starting point.
Because of the ressource constraints of the .Net MF I splitted the whole library in several parts. I made a separate assembly for every barcode.

**Reply by josesimoes** [reply]
Got you! I did look for the NETMF implementation in the main repo. Wrongly interpreted the mention to it in the readme and thought that it was deprecated and removed.

Gald that's still there! Would you consider a new repo with a version targeting NET nanoFramework?

**Reply by micjahn** [reply]
It depends. I have no clue what are the differences between .Net Micro and Nano framework. If there are only some minor differences then perhaps it is a good idea to integrate the Nano target in the .Net MF repo.
Anyway, the .Net MF is far behind the main repo because it is hard to maintain. The constraints of the .Net MF are very hard for the implementation. I stopped maintaining that port because I'm a one-man-show and it is time-consuming.
But if it makes sense in your eyes to build a nano version upon the MF repo and you are willing to provide the necessary pull requests I will gladly merge them.

**Reply by josesimoes** [reply]
nanoFramework started in 2016 as a spin off from NETMF (which is defunct for quite some time now). You can check more details on both github repos.

Having said that and considering your comments above, it would probably make more sense to have a port for nanoFramework based on the current ZXing code base.

To properly set expectations: this is not "business critical" to us just wanted to gauge your willingness on this and if it would something to consider or not. 🙂

**Reply by josesimoes** [reply]
Forgot to clarify that this is not a request for you to code and babysit the library. If this goes forward we're willing to commit time to maintain it. Just like what we've been doing with some other projects we've been cooperating with.
